### ISSUE TITLE
Track overruns should error

### ISSUE THREAD

**Issue by tdaede** [keirf/greaseweazle#247]
The current pre-index padding code ultimately results in skewing the clock rate if the sector data does not fit in the track length. It should instead exit with an error.

**Comment by keirf** [comment]
Which pre-index padding code?

**Comment by tdaede** [comment]
mfm.py:246

```
        # Add the pre-index gap.
        tlen = int((self.time_per_rev / self.clock) // 16)
        gap = max(tlen - len(t)//2, 0)
        t += encode(bytes([self.gapbyte] * gap))
```

The max() function results in it not padding if the track is too long, which then potentially results in the RPM correction code doing the wrong thing.

**Comment by keirf** [comment]
Yes, so a long track squishes the bits to fit. This can happen. Perhaps not in current uses of `IBM_MFM` and `IBM_FM` though.

**Comment by keirf** [comment]
Perhaps there should be an error or warning for gross misfits (eg clock gets squished by 20% or more)

**Comment by tdaede** [comment]
The current easiest way to trigger it is with D88 files that either need a smaller gap3, or were dumped from floppies with overlapping sectors (D88 can't represent the overlap, but it works on emulators that don't care about track length).

**Comment by keirf** [comment]
If D88 changes to use the IBMTrackFormat logic (same as IMD) then at least the disks with short gap3 will work. Overlapping sectors likely have a massively over large track and we could definitely warn on that.
